app/server/controllers/question.ts

```diff
@@ -1,17 +1,33 @@
 import * as AppInsights from 'app/server/app-insights';
-import { Router} from 'express';
+import { NextFunction, Request, Response, Router } from 'express'
 import * as Paths from 'app/server/paths';
 import { answerValidation } from 'app/server/utils/fieldValidation';
 
+export function getCurrentQuestion(req: Request) {
+  if (!req.session.questions) {
+    return undefined;
+  }
+  const questionOrdinal = parseInt(req.params.questionOrdinal, 10);
+  const currentQuestion = req.session.questions.find(q => q.question_ordinal === questionOrdinal);
+  if (!currentQuestion) {
+    return undefined;
+  }
+  return currentQuestion;
+}
+
 function getQuestion(getQuestionService) {
-  return async(req, res, next) => {
+  return async(req: Request, res: Response, next: NextFunction) => {
+    const currentQuestion = getCurrentQuestion(req);
+    if (!currentQuestion) {
+      return res.redirect(Paths.taskList);
+    }
     const hearingId = req.session.hearing.online_hearing_id;
-    const questionId = req.params.questionId;
     try {
-      const response = await getQuestionService(hearingId, questionId);
+      const response = await getQuestionService(hearingId, currentQuestion.question_id);
 
       const question = {
```

Comment:
This mapping should probably moved to the service?  (Separate PR)

Reply:
Not sure I understand what you mean for this one. This is just setting up the vars for the template?

Reply:
In the question service we return a type Question in the format that the template is expecting